src/darwin/Framework/CHIP/CHIPDevice.h

```diff
@@ -134,11 +163,10 @@ extern NSString * const kCHIPStatusKey;
 /**
  * Subscribe an attribute in a designated attribute path
  *
- * @param reportHandler   handler for the reports. Note that only the report handler by the last call to this method per the same
- * attribute path will receive reports. Report handler will receive either value or error. value will be an NSDictionary object. The
- * NSDictionary object will have "endpointId", "clusterId", "attributeId" and "value" keys. "endpointId", "clusterId" and
- * "attributeId" will be mapped to NSNumber objects. "value" key value will be an NSDictionary object representing attribute value
- *                      as described in the readAttributeWithEndpointId:clusterId:attributeId:clientQueue:responseHandler: method.
+ * @param reportHandler   handler for the reports. Report handler will receive either value or error.
+ *
+ *                    Received value is a response-value NSDictionary object as described in
+ *                    readAttributeWithEndpointId:clusterId:attributeId:clientQueue:completion:.
  */
 - (void)subscribeAttributeWithEndpointId:(NSUInteger)endpointId
```

Comment:
Could the `reportHandler` argument here also be of type `CHIPDeviceResponseHandler`?

Reply:
Yes, good point. It makes sense to have a consistent interface.